### ISSUE TITLE
Search follow symlinks spawns an rg process that uses 100% of CPU

### ISSUE THREAD

**Issue by scandar** [microsoft/vscode#206482]
Does this issue occur when all extensions are disabled?: Yes

Version: 1.86.2 (Universal)
Commit: 903b1e9d8990623e3d7da1df3d33db3e42d80eda
Date: 2024-02-13T19:42:13.651Z
Electron: 27.2.3
ElectronBuildId: 26908389
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Darwin arm64 23.2.0

Steps to Reproduce:

1. Make sure `Search > Follow symlinks` option is enabled in settings
2. Open a big enough repo (in my case it's a monorepo that uses pnpm and has a lot of symlinks in node_modules)
3. Use the global find and replace to look for something
4. after a while you'll see that your CPU usage is going up to 100% and most of it is consumed by `rg` process

The temporary fix is to disable `Follow symlinks`, but this option is enabled by default and pretty much everyone who works with this repo has the same problem.

**Comment by kmanwar89** [comment]
I'm having this same issue, also as previously described in #98594; details below:

Does this issue occur when all extensions are disabled?: No (but see details below)

Version: 1.87.1
Commit: 1e790d77f81672c49be070e04474901747115651
Date: 2024-03-06T00:21:07.287Z
Electron: 27.3.2
ElectronBuildId: 26836302
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Linux x64 6.5.0-18-generic (ZorinOS, based on Ubuntu 22.04LTS)

I've isolated it to, specifically, the **Microsoft** published-extension (ms-vscode-remote.remote-containers) "Dev Containers".  As soon as I enable this extension, all 8 cores of my i7-6700k CPU go to 100% sustained, as shown in `bashtop`. The process taking up the most CPU time (~50% each) is `rg` (ripgrep), which searching indicates "ripgrep recursively searches directories for a regex pattern while respecting your gitignore", with their [GitHub page](https://github.com/BurntSushi/ripgrep) linked here.

I've included a screen recording clearly demonstrating this behavior:

https://github.com/microsoft/vscode/assets/11323060/d6e5b762-d534-4170-a0ad-917d514481f6

**Comment by scandar** [comment]
@kmanwar89 
> I've isolated it to, specifically, the Microsoft published-extension (ms-vscode-remote.remote-containers) "Dev Containers".

So that's probably why I can't reproduce this problem anymore, since I deleted this extension.
Thanks for adding context to the problem.

**Comment by kmanwar89** [comment]
Cheers, thanks for creating this issue so I didn't have to :)

**Comment by andreamah** [comment]
@chrmarti This might be because you use the `findFiles` API, which will follow symlinks if the user sets `followSymlinks` to true. If you use `findFiles2`, which is proposed, you can force disable `followSymlinks` on your particular search.

**Comment by kmanwar89** [comment]
Hi, is there any update that can be shared on this issue?

**Comment by chrmarti** [comment]
@kmanwar89 Could you try the Dev Containers extension version 0.356.0-pre-release? That comes with a fix for microsoft/vscode-remote-release#9738.

Skipping symlinks might still be an option, but it would also miss devcontainer.json files in linked subfolders of the `.devcontainer` folder and links to devcontainer.json files themselves. 🤔

**Comment by kmanwar89** [comment]
> @kmanwar89 Could you try the Dev Containers extension version 0.356.0-pre-release? That comes with a fix for [microsoft/vscode-remote-release#9738](https://github.com/microsoft/vscode-remote-release/issues/9738).
> 
> Skipping symlinks might still be an option, but it would also miss devcontainer.json files in linked subfolders of the `.devcontainer` folder and links to devcontainer.json files themselves. 🤔

Happy to test - could you share guidance on how might I get access to pre-release?

**Comment by chrmarti** [comment]
There is a `Switch to Pre-Release Version` button when you open the extension's description from the Extensions side-panel:

![Image](https://github.com/microsoft/vscode/assets/9205389/5de0b0ed-fbab-4c1d-8953-44a467d6b51d)

**Comment by kmanwar89** [comment]
Hi, I switched over tot the pre-release version and it appears to have cleared up the issue - monitoring my CPU doesn't show any significant increase with the extension enabled, and no noticeable decrease with it disabled.

I'll keep testing, because what I'm noticing is my base CPU usage seems a bit higher overall - ~40% while idle compared to <20% in the screencast that I shared previously. It's unclear what is causing this much of a difference since my setup hasn't changed significantly between posts, but I'll share any feedback if the issue reoccurs.
